src/main/java/sc/fiji/coloc/results/PDFWriter.java

```diff
@@ -102,19 +104,85 @@ public void handleImage(RandomAccessibleInterval<T> image, String name) {
 	public void handleHistogram(Histogram2D<T> histogram, String name) {
 		RandomAccessibleInterval<LongType> image = histogram.getPlotImage();
 		ImagePlus imp = ImageJFunctions.wrapFloat( image, name );
+		
 		// make a snapshot to be able to reset after modifications
-		imp.getProcessor().snapshot();
+		//imp.getProcessor().snapshot();
 		imp.getProcessor().log();
 		imp.updateAndDraw();
 		imp.getProcessor().resetMinAndMax();
 		IJ.run(imp,"Fire", null);
+		
+		Overlay overlay = new Overlay();
+		
+		/*
+		 * check if we should draw a regression line for the current
+		 * histogram.
+		 */
+		if (histogram.getDrawingSettings().contains(Histogram2D.DrawingFlags.RegressionLine)) {
+			AutoThresholdRegression<T> autoThreshold = this.container.getAutoThreshold();
+			if (histogram != null && autoThreshold != null) {
+				drawLine(histogram, overlay, image.dimension(0), image.dimension(1),
+						autoThreshold.getAutoThresholdSlope(), autoThreshold.getAutoThresholdIntercept());
+				overlay.setStrokeColor(java.awt.Color.WHITE);
+				imp.setOverlay(overlay);
+			}
+		}
+		
 		addImageToList(imp, name);
 		// reset the imp from the log scaling we applied earlier
 		imp.getProcessor().reset();
```

Comment:
The `snapshot` and the `reset` go together. If one is commented-out/removed, they both should be.

Reply:
The commit has been amended accordingly.